eox_tenant/backends/database.py

```diff
@@ -180,3 +180,156 @@ def set_key_to_cache(self, key, value):
             self.current_request_configuration.cache = {}
 
         self.current_request_configuration.cache[key] = value
+
+
+class TenantConfigCompatibleMicrositeBackend(EdunextCompatibleDatabaseMicrositeBackend):
+    """
+    Backend that reads the configurations definitions from the database
+    using the custom models.
+    """
+
+    _backend_manager = None
+
+    @property
+    def backend_manager(self):
+        """
+        Return eox_tenant microsite manager.
+        """
+        if not self._backend_manager:
+            from eox_tenant.models import TenantConfig
+            self._backend_manager = TenantConfig
+        return self._backend_manager
+
+    def iterate_sites_with_tenant(self):
+        """
+        Return all the microsites from the database storing the results in the current request to avoid
+        quering the DB multiple times in the same request.
+        """
+
+        cache_key = "tenant-microsites-iterator"
+        cached_list = self.get_key_from_cache(cache_key)
+
+        if cached_list:
+            candidates = cached_list
+        else:
+            candidates = self.backend_manager.objects.all()  # pylint: disable=no-member
+            self.set_key_to_cache(cache_key, candidates)
+
+        for microsite in candidates:
```

Comment:
The purpose is to iterate over the cached query, avoiding extra hits to the database. Also, it is helper method since we must iterate over all configs in other places.